src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -670,7 +794,24 @@ private void EmitModule(JsonTextWriter jsonWriter, ModuleSymbol moduleSymbol, Ex
                 // If it is a template spec module, emit templateLink instead of template contents.
                 jsonWriter.WritePropertyName(moduleSemanticModel is TemplateSpecSemanticModel ? "templateLink" : "template");
                 {
-                    TemplateWriterFactory.CreateTemplateWriter(moduleSemanticModel, this.settings).Write(jsonWriter);
+                    var nestedTemplateOffset = jsonWriter.CurrentPos;
+
+                    var writer = TemplateWriterFactory.CreateTemplateWriter(moduleSemanticModel, this.settings);
+                    writer.Write(jsonWriter);
+
+                    if (writer is TemplateWriter templateWriter && moduleSemanticModel is SemanticModel moduleSemanticModelObj)
+                    {
+                        var childAbsolutePath = moduleSemanticModelObj.SourceFile.FileUri.AbsolutePath;
+                        var parentAbsolutePath = Path.GetDirectoryName(this.context.SemanticModel.SourceFile.FileUri.AbsolutePath)!;
+                        var modulePathRelativetoParent = Path.GetRelativePath(
+                            parentAbsolutePath,
+                            childAbsolutePath);
```

Comment:
Create a utility method for this - add to `IFileResolver`. We want to stay OS-agnostic, so ensure you use `/` instead of `\` for the path separator on all platforms.